server/src/main/java/org/elasticsearch/action/admin/indices/get/GetIndexResponse.java

```diff
@@ -156,5 +224,202 @@ public void writeTo(StreamOutput out) throws IOException {
+            if (indexDefaultSettings != null) this.indexDefaultSettings = indexDefaultSettings;
+        }
+    }
+
+    public static GetIndexResponse fromXContent(XContentParser parser) throws IOException {
+        ImmutableOpenMap.Builder<String, List<AliasMetaData>> aliases = ImmutableOpenMap.builder();
+        ImmutableOpenMap.Builder<String, ImmutableOpenMap<String, MappingMetaData>> mappings = ImmutableOpenMap.builder();
+        ImmutableOpenMap.Builder<String, Settings> settings = ImmutableOpenMap.builder();
+        ImmutableOpenMap.Builder<String, Settings> defaultSettings = ImmutableOpenMap.builder();
+        List<String> indices = new ArrayList<>();
+
+        if (parser.currentToken() == null) {
+            parser.nextToken();
+        }
+        ensureExpectedToken(Token.START_OBJECT, parser.currentToken(), parser::getTokenLocation);
+        parser.nextToken();
+
+        while (!parser.isClosed()) {
+            if (parser.currentToken() == Token.START_OBJECT) {
+                // we assume this is an index entry
+                String indexName = parser.currentName();
+                indices.add(indexName);
+                IndexEntry indexEntry = parseIndexEntry(parser);
+                // make the order deterministic
+                CollectionUtil.timSort(indexEntry.indexAliases, Comparator.comparing(AliasMetaData::alias));
+                aliases.put(indexName, Collections.unmodifiableList(indexEntry.indexAliases));
+                mappings.put(indexName, indexEntry.indexMappings);
+                settings.put(indexName, indexEntry.indexSettings);
+                if (indexEntry.indexDefaultSettings.isEmpty() == false) {
+                    defaultSettings.put(indexName, indexEntry.indexDefaultSettings);
+                }
+            } else if (parser.currentToken() == Token.START_ARRAY) {
+                parser.skipChildren();
+            } else {
+                parser.nextToken();
+            }
+        }
+        return
+            new GetIndexResponse(
+                indices.toArray(new String[0]), mappings.build(), aliases.build(),
+                settings.build(), defaultSettings.build()
+            );
+    }
+
+    @Override
+    public String toString() {
+        return Strings.toString(this);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o== null || getClass() != o.getClass()) return false;
+        GetIndexResponse that = (GetIndexResponse) o;
+        return Arrays.equals(indices, that.indices) &&
+            Objects.equals(aliases, that.aliases) &&
+            Objects.equals(mappings, that.mappings) &&
+            Objects.equals(settings, that.settings) &&
+            Objects.equals(defaultSettings, that.defaultSettings);
+    }
+
+    @Override
+    public int hashCode() {
+        return
+            (31 * Arrays.hashCode(indices)) +
+                Objects.hash(
```

Comment:
I believe it should be safe to do `Objects.hash(Arrays.hashCode(indices), aliases, mappings, settings, defaultSettings)`

Reply:
I believe so as well. I don't remember why I did it like this anymore. I will make the change.